src/components/form/date-input/date-input.component.ts

```diff
@@ -173,9 +174,23 @@ export class DateInputComponent implements ControlValueAccessor, Validator, OnIn
     this.propagateChange(this.rawValue);
   }
 
+  public inputFocus() {
+    this.isTouched = false;
+    this.touch();
+  }
+
   public inputBlur() {
-    this.formControl.markAsTouched();
-    this.propagateChange(this.rawValue);
+    this.isTouched = true;
+    this.touch();
+  }
+
+  public touch() {
```

Comment:
I don't see touch() being called anywhere in the git diff, so wouldn't:
```
public inputFocus() {
    this.formControl.markAsUntouched();
}

public inputBlur() {
    this.formControl.markAsTouched();
    this.propagateChange(this.rawValue);
}
```
Do the same thing?

Reply:
touch() is called at Line 184 in date-input.component.ts